mona_sdk/authentication.py

```diff
@@ -121,29 +124,44 @@ def _get_error_string_from_token_info(api_key):
     return ", ".join(_get_token_info_by_api_key(api_key, ERRORS)) if error_list else ""
 
 
-def _request_access_token_with_retries(api_key, secret):
+def _request_access_token_with_retries(client):
     return _get_auth_response_with_retries(
-        lambda: _request_access_token_once(api_key, secret)
+        lambda: _request_access_token_once(client.api_key, client.secret),
+        num_of_retries=client.num_of_retries_for_authentication,
+        auth_wait_time_sec=client.wait_time_for_authentication_retries,
     )
 
 
-def _request_refresh_token_with_retries(refresh_token_key):
+def _request_refresh_token_with_retries(refresh_token_key, client):
     return _get_auth_response_with_retries(
-        lambda: _request_refresh_token_once(refresh_token_key)
+        lambda: _request_refresh_token_once(refresh_token_key),
+        num_of_retries=client.num_of_retries_for_authentication,
+        auth_wait_time_sec=client.wait_time_for_authentication_retries,
     )
 
 
 def _get_auth_response_with_retries(
     response_generator,
-    num_of_retries=NUM_OF_RETRIES_FOR_AUTHENTICATION,
-    auth_wait_time_sec=WAIT_TIME_FOR_AUTHENTICATION_RETRIES_SEC,
+    num_of_retries=None,
+    auth_wait_time_sec=None,
 ):
     """
     Sends an authentication request (first time/refresh) with a retry mechanism.
     :param response_generator (lambda)
             A function call that sends the wanted REST request.
     :return: The response received from the authentication server.
     """
+    num_of_retries = (
```

Comment:
Why did you make this change (I mean all changes in _get_auth_response_with_retries)? It seems to have worked the same before - the default in the function definition is not used if caller passes anything to it anyway, am I missing something?

Reply:
The thing is that it always pass something (None is also something), I tried it as you said and since the function got None argument It didn't set the env vars value (its only setting the optional value if the function call did not pass anything, not even None)

Reply:
I didn't find a nicer way to do that, if you have an idea i'd love to hear I don't think it looks super fine.